File: week-04/day-4/09.py
# 9. Given a string, compute recursively a new string where all the
# adjacent chars are now separated by a "*".


text = "abcde"

def coder(text):
    if len(text) == 1:
        return text
    else:
        return text[0] + "*" + coder(text[1::])

print (coder(text))

File: week-07/day-1/recursion/09.js
// 9. Given a string, compute recursively a new string where all the
// adjacent chars are now separated by a "*".

var text = "abcde"

function asterix(inputText) {
  if (inputText.length == 1) {
    return inputText;
  } else {
    return inputText[0] + '*' + asterix(inputText.slice(1, inputText.length));
  }
}

console.log(asterix(text))
